### ISSUE TITLE
Handling import aliases in macro

### ISSUE THREAD

**Issue by kLabz** [HaxeFoundation/haxe#7489]
I have a macro that does some type checking / enforcing on typedefs located in other modules. Sometimes these typedefs can use types imported with aliases (I have no control over what is done there), and this doesn't work (tested with haxe 3.4.4, 3.4.7, 4 preview 4 and dev).

It can be reduced to this:

#### `Test.hx`
```haxe
import String as StringAlias;

class Test {
  static function main() {
    Macro.main();
  }
}

typedef A = {
  var a:StringAlias->Void;
}
```

#### `Macro.hx`
```haxe
class Macro {
  public static function main() {
    var a:Test.A = {a: function(_) {}};

    // Macro.hx:6: characters 9-12 : Type not found : StringAlias
    test(a.a);
  }
  
  public static macro function test(e) {
    var t = haxe.macro.Context.typeof(e);

    // Doesn't help, but it would if we had `StringAlias` instead of `StringAlias->Void`
    // t = haxe.macro.Context.follow(t);

    var ct = haxe.macro.TypeTools.toComplexType(t);
    return macro @:pos(e.pos) ($e :$ct);
  }
}
```

[Here](http://try-haxe.mrcdk.com/#6822F)'s a try-haxe with the issue and some variations commented out (which have different behavior regarding error reporting with 3.4.4 and dev).

The only workaround I've found so far is catching the exception and telling the developer to use `typedef StringAlias = String` instead of `import String as StringAlias`.

Is there a way to make this work? Is this considered a bug?

**Comment by Simn** [comment]
I'm not sure if this isn't the expected behavior. You return the complex type `StringAlias` from your macro and the compiler then has no idea how to handle that in the context of your `Macro` class, which is the call-site.

You can `t = t.map(t -> t.follow());` (with `using haxe.macro.Tools`) to follow the type recursively. Otherwise, `follow` stops at the `TFun` because there's nothing to follow there.

**Comment by Simn** [comment]
Also you cannot try/catch anything here because the error doesn't happen in the `typeof` or `toComplexType` calls. It occurs while typing the result of the (successful) macro call.

**Comment by kLabz** [comment]
Found [a way](http://try-haxe.mrcdk.com/#16d1E) to handle all these cases (map wasn't working for the commented out cases), thanks!

I'm not sure about the way Haxe handles this though, something seems odd. Especially since the workaround I found was to use `typedef StringAlias = String` instead of `import String as StringAlias` which are not the same things but (I guess) are often regarded as doing more or less the same thing.

I'm not closing this, but feel free to do so if you think nothing's wrong with the way it works / gives errors right now :)

Thanks again!
